Bluetooth: Mesh: LPN terminate CB establish clause

Andrewpini opened the pull request:
Adds clause so that the LPN must have a established friend
connection for the connection terminate callback to trigger
upon clearing a friendship.

Also adds BabbleSim test covering this particular corner case.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

Balaklaka commented on tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c:
```diff
@@ -815,6 +815,26 @@ static void test_other_group(void)
+	k_sleep(K_MSEC(400));
+	bt_mesh_lpn_set(false);
+
+	if (!evt_wait(LPN_TERMINATED, K_SECONDS(30))) {
```
Is it necessary to wait 30 seconds? Could the wait time be shorter (5 or 10 sec), or this test will take more than 30 sec to execute.

Andrewpini replied:
No, I guess it could be shorter. 30 Sec seems to be a commoly used period. Since it is simulated time I guess that the impact is minimal regardless?

trond-snekvik commented on subsys/bluetooth/mesh/lpn.c:
```diff
@@ -240,9 +240,11 @@ static void clear_friendship(bool force, bool disable)
-	STRUCT_SECTION_FOREACH(bt_mesh_lpn_cb, cb) {
-		if (cb->terminated && lpn->frnd != BT_MESH_ADDR_UNASSIGNED) {
-			cb->terminated(lpn->sub->net_idx, lpn->frnd);
+	if ((lpn->frnd != BT_MESH_ADDR_UNASSIGNED) && lpn->established) {
```
```suggestion
	if (lpn->established) {
```
is sufficient, I believe?

Andrewpini replied:
Yes, I guess 👍

PavelVPV commented on tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c:
```diff
@@ -815,6 +815,26 @@ static void test_other_group(void)
+	bt_mesh_test_setup();
+
+	bt_mesh_lpn_set(true);
+	k_sleep(K_MSEC(400));
```
Could this be replaced with `evt_wait(LPN_POLLED, K_MSEC(...))`? Timeout could be some reasonable value, for example, number of Friend Requests including delay for response.

Andrewpini committed:
Bluetooth: Mesh: Bsim test for LPN terminate cb

Test covering a corner case scenario where the LPN has received a
friend offer, but has not yet established a connection. In this case
the LPN terminate callback should not be triggered if the LPN is
disabled, which is monitored by this test.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

Andrewpini committed:
Bluetooth: Mesh:  LPN teminate cb establish clause

Adds clause so that the LPN  must have a established friend
connection for the connection terminate callback to trigger
upon clearing a friendship.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>
